Makefile

```diff
@@ -1,21 +1,32 @@
-.PHONY: build-rust-optimizer build-workspace-optimizer build publish-rust-optimizer publish-workspace-optimizer publish
+.PHONY: build-rust-optimizer build-workspace-optimizer build create-rust-optimizer-builder use-rust-optimizer-builder publish-rust-optimizer publish-workspace-optimizer publish
 
 DOCKER_NAME_RUST_OPTIMIZER := "cosmwasm/rust-optimizer"
 DOCKER_NAME_WORKSPACE_OPTIMIZER := "cosmwasm/workspace-optimizer"
-DOCKER_TAG := 0.12.1
+DOCKER_TAG := 0.12.2
+
+# Build images locally for the host CPU architecture
 
 build-rust-optimizer:
 	docker build -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --target rust-optimizer .
 
 build-workspace-optimizer:
 	docker build -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --target workspace-optimizer .
 
-publish-rust-optimizer: build-rust-optimizer
-	docker push $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG)
+build: build-rust-optimizer build-workspace-optimizer
 
-publish-workspace-optimizer: build-workspace-optimizer
-	docker push $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG)
+# Build multi-CPU architecture images and publish them. rust alpine images support the linux/amd64 and linux/arm64/v8 architectures.
 
-build: build-rust-optimizer build-workspace-optimizer
+create-rust-optimizer-builder:
```

Comment:
The requirement to run `create-rust-optimizer-builder` is suboptimal. Maybe we could try running `use-rust-optimizer-builder` and if it fails, run `create-rust-optimizer-builder`?

Reply:
Can't we use one of those standard builders? Why do we need our own?

```
docker buildx ls
NAME/NODE       DRIVER/ENDPOINT STATUS  PLATFORMS
desktop-linux   docker                  
  desktop-linux desktop-linux   running linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
default *       docker                  
  default       default         running linux/amd64, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
```

Reply:
Good point. I didn't knew they were there.

Reply:
The default builder doesn't work on my M1

```
❯ docker buildx use default
❯ docker buildx build --platform linux/amd64,linux/arm64/v8 .
[+] Building 0.0s (0/0)
error: multiple platforms feature is currently not supported for docker driver. Please switch to a different driver (eg. "docker buildx create --use")
```

Reply:
Thanks. I Implemented that ichange in my spin-off branch, but didn't tested it very well. Will revert.